mmv1/third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb

```diff
@@ -83,12 +84,115 @@ func resourceDataflowFlexTemplateJob() *schema.Resource {
+				Description: `The maximum number of Google Compute Engine instances to be made available to your pipeline during execution, from 1 to 1000.`,
+			},
+
+			"service_account_email": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The Service Account email used to create the job.`,
+			},
+
+			"temp_location": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The Cloud Storage path to use for temporary files. Must be a valid Cloud Storage URL, beginning with gs://.`,
+			},
+
+			"staging_location": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The Cloud Storage path to use for staging files. Must be a valid Cloud Storage URL, beginning with gs://.`,
+			},
+
+			"sdk_container_image": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The Cloud Storage path to use for temporary files. Must be a valid Cloud Storage URL, beginning with gs://.`,
+			},
+
+			"network": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+				Description:      `The network to which VMs will be assigned. If it is not provided, "default" will be used.`,
+			},
+
+			"subnetwork": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+				Description:      `The subnetwork to which VMs will be assigned. Should be of the form "regions/REGION/subnetworks/SUBNETWORK".`,
+			},
+
+			"machine_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The machine type to use for the job.`,
+			},
+
+			"kms_key_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The name for the Cloud KMS key for the job. Key format is: projects/PROJECT_ID/locations/LOCATION/keyRings/KEY_RING/cryptoKeys/KEY`,
+			},
+
+			"ip_configuration": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  `The configuration for VM IPs. Options are "WORKER_IP_PUBLIC" or "WORKER_IP_PRIVATE".`,
+				ValidateFunc: validation.StringInSlice([]string{"WORKER_IP_PUBLIC", "WORKER_IP_PRIVATE", ""}, false),
+			},
+
+			"additional_experiments": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Description: `List of experiments that should be used by the job. An example value is ["enable_stackdriver_agent_metrics"].`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"enable_streaming_engine": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: `Indicates if the job should use the streaming engine feature.`,
+			},
+
+			"autoscaling_algorithm": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The algorithm to use for autoscaling`,
+			},
+	
+			"launcher_machine_type": {
```

Comment:
The API docs mention a default of `n1-standard-1`. Do you know if this is returned by the server, and perhaps we want to use `Computed: true`?

Reply:
It is defined as an option simalar as machine_type https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.locations.flexTemplates/launch#flextemplateruntimeenvironment. not something that is computed though?

Reply:
Sorry, could you clarify? Is the machine type set by the server but not reflected in the field? I know that you're copying from existing behavior, but I think it would be good to know how it is working.